src/Workspaces/Core/Portable/RegularExpressions/RegexCharClass.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.RegularExpressions
+{
+    /// <summary>
+    /// Minimal copy of https://github.com/dotnet/corefx/blob/master/src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs
+    /// Used to accurately determine if something is a WordChar according to the .Net regex engine.
+    /// </summary>
+    internal static class RegexCharClass
```

Comment:
this code was copied over near verbatim.  All i did was clean up formatting and remove any methods that were not needed at all by the new regex parser.